### PR TITLE
status messages for cockpit

### PR DESCRIPTION
**Pull request by sub-mod**
add Custom log messages for Cockpit to show progress of atomicapp command.
1)for status
2)for answer file contents
3)for error messages

### CODE REVIEW

File: atomicapp/utils.py [hunk 1]
```diff
@@ -14,6 +14,18 @@
 
 logger = logging.getLogger(__name__)
 
+def printStatus(message):
+    logger.info("atomicapp.status.message="+str(message))
```
Comment (goern) [inline atomicapp/utils.py]: why is this not atomicapp.status.info.message?

File: atomicapp/utils.py [hunk 2]
```diff
@@ -14,6 +14,18 @@
...
+
+
+def printErrorStatus(message):
+    logger.info("atomicapp.status.error.message="+str(message))
```
Comment (goern) [inline atomicapp/utils.py]: and why is always the logger.info used?

File: atomicapp/utils.py [hunk 3]
```diff
@@ -121,7 +133,12 @@ def askFor(what, info):
                 if len(value) == 0:
                     value = info["default"]
             else:
-                value = raw_input("%s (%s): " % (what, desc))
+                try:
```
Comment (goern) [inline atomicapp/utils.py]: it this really part of the cockpit integration?

**Comment by goern** [comment]
regardless of the fact that your PR is missing a description and test cases, I have put some comments in here...

File: atomicapp/cli/main.py [hunk 4]
```diff
@@ -9,16 +9,19 @@
...
 from atomicapp import set_logging
 from atomicapp.constants import ANSWERS_FILE, __ATOMICAPPVERSION__, __NULECULESPECVERSION__
+from atomicapp.utils import printStatus
 
 logger = logging.getLogger(__name__)
 
 def cli_install(args):
     install = Install(**vars(args))
     install.install()
+    printStatus("Install Successful.")
```
Comment (goern) [inline atomicapp/cli/main.py]: shall we have machine readable and human readable status messages that are send back to cockpit? @stefwalter

File: atomicapp/install.py [hunk 5]
```diff
@@ -65,12 +66,14 @@ def _copyFromContainer(self, image):
         logger.debug("Creating a container with name %s", name)
 
         create = [self.docker_cli, "create", "--name", name, image, "nop"]
+        logger.debug(" ".join(create))
         subprocess.call(create)
         cp = [self.docker_cli, "cp", "%s:/%s" % (name, APP_ENT_PATH), self.utils.tmpdir]
-        logger.debug(cp)
+        logger.debug(" ".join(cp))
```
Comment (goern) [inline atomicapp/install.py]: this seems to be unrelated to cockpit integration?! same for line 5

File: atomicapp/providers/kubernetes.py [hunk 6]
```diff
@@ -70,7 +70,16 @@ def _callK8s(self, path):
...
-            subprocess.check_call(cmd)
+            try:
+                p = Popen(cmd, stdout=PIPE, stderr=PIPE)
+                stdout, stderr = p.communicate()
+                logger.debug("stdout = %s", stdout)
+                logger.debug("stderr = %s", stderr)
+                if stderr and  stderr.strip() != "":
+                    raise Exception(str(stderr))
+            except Exception:
+                logger.error("cmd failed: %s", " ".join(cmd))
```
Comment (goern) [inline atomicapp/providers/kubernetes.py]: why not an additional printErrorStatus() so that cockpit knows whats going on?

File: atomicapp/run.py [hunk 7]
```diff
@@ -110,8 +113,12 @@ def _applyTemplate(self, data, component):
             except KeyError as ex:
                 name = ex.args[0]
                 logger.debug("Artifact contains unknown parameter %s, asking for it", name)
-                config[name] = self.utils.askFor(name, {"description": "Missing parameter '%s', provide the value or fix your %s" % (name, MAIN_FILE)})
+                try:
```
Comment (goern) [inline atomicapp/run.py]: this whole block looks confusing, @vpavlin pls have a look at it too

File: atomicapp/install.py (hunk 5 above)
Comment (sub-mod) [inline atomicapp/install.py]: is it ok to have this logger message to print this docker command ?

File: atomicapp/providers/kubernetes.py (hunk 6 above)
Comment (sub-mod) [inline atomicapp/providers/kubernetes.py]: ok added.

File: atomicapp/utils.py (hunk 1 above)
Comment (sub-mod) [inline atomicapp/utils.py]: ok done.

File: atomicapp/run.py (hunk 7 above)
Comment (sub-mod) [inline atomicapp/run.py]: yes this is added for cockpit integration. I want askFor method to give a user defined exception to cockpit.

File: atomicapp/utils.py (hunk 2 above)
Comment (sub-mod) [inline atomicapp/utils.py]: I dont have any particular preference for info/debug log levels here. any acceptable log level is ok for me.

File: atomicapp/utils.py (hunk 3 above)
Comment (sub-mod) [inline atomicapp/utils.py]: yes

File: atomicapp/install.py (hunk 5 above)
Comment (goern) [inline atomicapp/install.py]: sure, but dont put it in this PR than

**Comment by goern** [comment]
thanks for fixing the comments, pls have a look at landscape's comment too :) https://landscape.io/diff/195898

File: atomicapp/run.py [hunk 8]
```diff
@@ -110,8 +113,12 @@ def _applyTemplate(self, data, component):
...
+                    config[name] = self.utils.askFor(name, {"description": "Missing parameter '%s', provide the value or fix your %s" % (name, MAIN_FILE)})
+                except EOFError:
```
Comment (vpavlin) [inline atomicapp/run.py]: Why do we need to check and log same exception twice? (here and https://github.com/projectatomic/atomicapp/pull/169/files#diff-1e107d733a5bd148a19592d4709e33a8R138)

File: atomicapp/run.py (hunk 7 above)
Comment (vpavlin) [inline atomicapp/run.py]: Why do we need to check and log same exception twice? (on line 118 and https://github.com/projectatomic/atomicapp/pull/169/files#diff-1e107d733a5bd148a19592d4709e33a8R138)

**Comment by vpavlin** [comment]
Could we somehow not to duplicate so many log messages?

https://github.com/projectatomic/atomicapp/pull/169/files#diff-b5e948be30e6739a24b1baa2c1132a2dR260
https://github.com/projectatomic/atomicapp/pull/169/files#diff-418e5fb2b4564e128feb32b96ed7855cR81
...

Printing those things twice as INFO doesn't sound completely right. Even if we agree it's ok, we can print it twice in `printStatus` function, not having to maintain 2 messages explicitly on random places in the code. 

Could you take a look at logging handlers (https://docs.python.org/2/library/logging.html#handler-objects) and think about how we could use that for (not only) cockpit instead of logging to stdout?

File: atomicapp/install.py (hunk 5 above)
Comment (sub-mod) [inline atomicapp/install.py]: ok removed it.

File: atomicapp/run.py (hunk 7 above)
Comment (sub-mod) [inline atomicapp/run.py]: that was an oversight, i have removed the extra log message.

**Comment by sub-mod** [comment]
@vpavlin @goern 
I have removed the 2 instances of duplication of log messages 
I took a look at adding custom handlers, but that doesnt help in my case. I have unique logger messages which I need for cockpit.

File: atomicapp/install.py [hunk 9]
```diff
@@ -159,12 +168,14 @@ def _installDependencies(self):
             mainfile_component_path = os.path.join(component_path, MAIN_FILE)
             logger.debug("Component path: %s", component_path)
             if not os.path.isfile(mainfile_component_path) or self.nulecule_base.update:
-                logger.info("Pulling %s", image_name)
+                printStatus("Pulling %s ..." % image_name)
                 component_app = Install(self.nulecule_base.answers_data, image_name, self.nulecule_base.nodeps, 
                                         self.nulecule_base.update, component_path, self.dryrun)
                 values = Utils.update(values, component_app.install())
+                printStatus("Component %s installed successfully."%component)
```
Comment (sub-mod) [inline atomicapp/install.py]: I cant help duplication here because i dont want to show the path in cockpit.

File: atomicapp/install.py [hunk 10]
```diff
@@ -159,12 +168,14 @@ def _installDependencies(self):
...
                 logger.info("Component installed into %s", component_path)
             else:
+                printStatus("Component %s already installed."%component)
```
Comment (sub-mod) [inline atomicapp/install.py]: I cant help duplication here because i dont want to show the path in cockpit.

**Comment by goern** [comment]
LGTM @vpavlin

File: atomicapp/install.py (hunk 9 above)
Comment (vpavlin) [inline atomicapp/install.py]: Ok, then, please, change the `logger.info` with path to `logger.debug`

Comment (sub-mod) [inline atomicapp/install.py]: ok done

**Comment by sub-mod** [comment]
Thanks @vpavlin
